Dockerfile.ci-rp

```diff
@@ -0,0 +1,38 @@
+ARG REGISTRY
+FROM ${REGISTRY}/ubi8/go-toolset:1.20.10 as builder
+
+USER root
+ENV GOPATH=/root/go
+RUN mkdir -p /app
+WORKDIR /app
+COPY . /app
+RUN make aro RELEASE=${IS_OFFICIAL_RELEASE} -o generate
+
+FROM builder as vendor-check
+WORKDIR /app
+COPY --from=builder /app/aro /app/e2e.test /usr/local/bin/
+RUN go mod vendor && go mod tidy -compat=1.18 && hack/ci-utils/isClean.sh
+
+FROM builder as lint-go
+WORKDIR /app
+COPY --from=builder /app/aro /app/e2e.test /usr/local/bin/
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin v1.55.2
```

Comment:
Linting is important in the build process, but wouldn't be easier to download the linter locally, add `lint-go` in the `ci-rp` target ([line 79](https://github.com/Azure/ARO-RP/pull/3489/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R79)) instead of doing that in the Dockerfile? 
IMO it would simplify the _Dockefile_, and make better code separation.
I am not sure where eventually the `ci-rp` target will be called. But it would be nice to test this target in a GitHub CI, this way you test linting per every PR in a CI image (e.g., Ubuntu).

Reply:
What you suggest is our current process (`make lint-go` already exists and expects this CLI to be installed locally). However, this has limitations:

1. This linter isn't the only dependency required - there are close to a dozen, so setup for a new dev can get incredibly difficult
2. It is difficult to get 50+ engineers in sync when we do things like bump the version of any of these tools

This is where this change comes in. They hope to make `podman` the only local dependency required to build and test our codebase. All you need day 1 and day 100 is `podman`, and we can easily keep everyone in sync with all our build dependencies with a `git pull`. It also ensures that every step of our code checks (builds/tests/linting/etc) happen locally the same way they happen during build processes.

Reply:
One of the goals behind this change was to have all CI "Just work" on developer machines, moving the lint out of the container may cause issues between dev setups.

Reply:
>What you suggest is our current process (make lint-go already exists and expects this CLI to be installed locally)

Correct, I was referring to adding the `lint-go` (or [`test-go`](https://github.com/Azure/ARO-RP/blob/master/Makefile#L208)) before building the container as you can see below. This way the test is done locally and not in the _Dockerfile.ci-rp_. 

```
ci-rp: lint-go
	docker build -f Dockerfile.ci-rp --build-arg REGISTRY=$(REGISTRY) --no-cache=$(NO_CACHE) --skip-unused-stages=false --jobs=0
```

I am not sure how the above limitations are relevant to the suggested change, but I do agree that it is "safer" to do lint in the Dockerfile rather than in the makefile target.
If it will be called mainly using the makefile target, then it seems pretty safe IMO

> One of the goals behind this change was to have all CI "Just work" on developer machines, moving the lint out of the container may cause issues between dev setups.

I am unaware of the usual dev setups that you folks do, but I guess the building procedure (multiple "tests" in the build process prior to the [actual container build](https://github.com/Azure/ARO-RP/pull/3489/files#diff-d189eef96a9753a0ef4403084a625155b556091a2b336d98e20f24139b7ac3acR27-R32)) in the Dockerfile has been discussed. I just wanted to share my two cents here.

Reply:
>I am not sure where eventually the ci-rp target will be called. But it would be nice to test this target in a GitHub CI, this way you test linting per every PR in a CI image (e.g., Ubuntu)

When is the `ci-rp` target supposed to be called in dev workflow?

Reply:
The idea here is that everything everywhere can just run these build containers from a make target.  So this would be ran in CI running in ADO pipelines, developers locally, etc...